core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -107,42 +128,79 @@ class LogSegment private[log] (val log: FileRecords,
   }
 
   /**
-   * Append the given messages starting with the given offset. Add
-   * an entry to the index if needed.
+   * Append the given messages ending at the given last offset. Add an entry to the index if needed.
    *
    * It is assumed this method is being called from within a lock.
    *
    * @param largestOffset The last offset in the message set
    * @param largestTimestamp The largest timestamp in the message set.
    * @param shallowOffsetOfMaxTimestamp The offset of the message that has the largest timestamp in the messages to append.
    * @param records The log entries to append.
-   * @return the physical position in the file of the appended records
    */
   @nonthreadsafe
   def append(largestOffset: Long,
              largestTimestamp: Long,
              shallowOffsetOfMaxTimestamp: Long,
              records: MemoryRecords): Unit = {
+    append(largestOffset, largestTimestamp, shallowOffsetOfMaxTimestamp, records, allowOversizeIndexOffset = false)
+  }
+
+  /**
+   * NOTE: Read the function description carefully. For most cases, you might want to call
```

Comment:
Nit: in my opinion, there's no need to tell people to read things carefully. :) It's enough to explain that the other method is what one generally wants and why. Also, these are generally called `methods` instead of `functions`.